filter.go

```diff
@@ -0,0 +1,61 @@
+package otelutil
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/sdk/trace"
+)
+
+// if this function returns false, the span will be filtered out
+type SpanFilterFunc func(span trace.ReadOnlySpan) bool
+type SpanProcessorWrapper func(sp trace.SpanProcessor) trace.SpanProcessor
+
+type filteringSpanProcessor struct {
+	filterFn SpanFilterFunc
+	sp       trace.SpanProcessor
+}
+
+var _ trace.SpanProcessor = (*filteringSpanProcessor)(nil)
+
+func (f *filteringSpanProcessor) OnStart(parent context.Context, s trace.ReadWriteSpan) {
+	if !f.filterFn(s) {
+		return
+	}
+	f.sp.OnStart(parent, s)
+}
+
+func (f *filteringSpanProcessor) OnEnd(s trace.ReadOnlySpan) {
+	if !f.filterFn(s) {
+		return
+	}
+	f.sp.OnEnd(s)
+}
+
+func (f *filteringSpanProcessor) Shutdown(ctx context.Context) error {
+	return f.sp.Shutdown(ctx)
+}
+
+func (f *filteringSpanProcessor) ForceFlush(ctx context.Context) error {
+	return f.sp.ForceFlush(ctx)
+}
+
+func WithSpanFilter(filterFn SpanFilterFunc) SpanProcessorWrapper {
+	return func(sp trace.SpanProcessor) trace.SpanProcessor {
+		return &filteringSpanProcessor{
+			sp:       sp,
+			filterFn: filterFn,
+		}
+	}
+}
+
+func WithSpanFilterOnlyNames(names ...string) SpanProcessorWrapper {
+	nameSet := make(map[string]struct{}, len(names))
+	for _, name := range names {
+		nameSet[name] = struct{}{}
+	}
+
+	return WithSpanFilter(func(span trace.ReadOnlySpan) bool {
```

Comment:
Could we use `span.InstrumentationScope()` to filter the scope so I do not have to include every possible name that I want to include in the track if i want to have more spans?

If I understand it correctly, a scope can have many span names.

Reply:
The span name is the one that we set in the file, for example in this code, `httputil.CallTC` will be the name of the span. I think we can simply for now put those names in the filter name and later we can optimize and provide a better API and filter functions to expose more functionality.

<img width="666" alt="Screenshot 2024-08-21 at 2 06 59 PM" src="https://github.com/user-attachments/assets/741a9a41-68c2-49f7-bdd7-416ba2b34950">

Reply:
Also I have exposed the function WithSpanFilter, which gives us the ability to write a filter the way we want, whether we want to use span.Name or `InstrumentationScope().Name`

Reply:
In that case, could we have `WithSpanFilterOnlyScopeNames` in the library to make the filtering much simpler?